### PR TITLE
Added an example extractor: `is`.

### PR DESCRIPTION
**Pull request by mpark**

### CODE REVIEW

**Comment by smurzin** [comment]
We would need to do "optimistic" parsing and rollback and reparse as identifier sometimes. 

```c++
inspect (v) {
    extract(v) ! x: std::cout << "extractor pattern";
    x << "identifier";
}
```

In the example above, I'll consume first token of extractor pattern (`extract`) and will get an ambiguity (either identifier or extractor) so will have to parse "optimistically" till I either find token that can't be inside extractor or find `!` or `?`.

**Comment by mpark** [comment]
Hm... I think I thought we were clear since we segregated the expression pattern, but forgot that I'm letting _constant expression_ be in the extractor pattern directly without `^`. Interesting. I'll leave the parens alone for now 🤔

**Comment by mpark** [comment]
@smurzin: I'm not sure if you're saying this would be unimplementable or if you're just keeping an implementation strategy note. Do we need to rework the syntax?

**Comment by smurzin** [comment]
Technically it's not impossible, but requires sort of look-ahead, where after we've got an `identifier` token from stream we have two possibilities:
* it's an extractor pattern -> next token should be one of `(`, `!`, `?`
* it's just an identifier pattern -> next token must be smth that makes sense only in pattern syntax

**Comment by mpark** [comment]
@smurzin: okay, seems like we'll need more implementation experience. since this PR doesn't change anything design-wise, I'll merge this for now.

File: D1371R0.pandoc [hunk 1]
```diff
@@ -867,18 +867,37 @@ Let `c` be the _constant-expression_. The first form matches value `v`
...
+```
+
 For second form, let `e` be the result of a call to member `c.try_extract(v)`
 or else a non-member ADL-only `try_extract(c, v)`. It matches value `v` if `e`
 is contextually convertible to `bool`, evaluates to `true`,
 and _pattern_ matches `*e`.
 
 ```cpp
-struct {
+constexpr struct {
```
Comment (camio) [inline D1371R0.pandoc]: This `constexpr` is going to confuse people. Is it necessary? FFTI.

**Pull request merged**